### PR TITLE
6731 add new artifact and attribute types based on i leapp

### PR DESCRIPTION
**Pull request by markmckinnon**

### CODE REVIEW

File: bindings/java/doxygen/artifact_catalog.dox [hunk 1]
```diff
@@ -35,6 +35,23 @@ TSK_CARD_NUMBER (Credit card number)
...
+Activity on the system or from an application.  Example Usage is a mobile device being locked and unlocked.
+. 
+
+### REQUIRED ATTRIBUTES
+- TSK_DATETIME_START (When activity Started)
+    or 
+  TSK_DATETIME
+
+### OPTIONAL ATTRIBUTES
+- TSK_ACTIVITY_TYPE (Activity Type ie: On or Off)
```
Comment (rcordovano) [inline bindings/java/doxygen/artifact_catalog.dox:48]: "ie" should be "i.e."

Comment (markmckinnon) [reply bindings/java/doxygen/artifact_catalog.dox]: Fixed that and a typo in one of the attribute names.

File: bindings/java/doxygen/artifact_catalog.dox [hunk 2]
```diff
@@ -35,6 +35,23 @@ TSK_CARD_NUMBER (Credit card number)
 
 ---
 
+## TSK_ACTIVITY
+Activity on the system or from an application.  Example Usage is a mobile device being locked and unlocked.
```
Comment (rcordovano) [inline bindings/java/doxygen/artifact_catalog.dox:39]: Usage => usage

File: bindings/java/doxygen/artifact_catalog.dox [hunk 3]
```diff
@@ -35,6 +35,23 @@ TSK_CARD_NUMBER (Credit card number)
...
+. 
+
+### REQUIRED ATTRIBUTES
+- TSK_DATETIME_START (When activity Started)
```
Comment (rcordovano) [inline bindings/java/doxygen/artifact_catalog.dox:43]: Started => started

File: bindings/java/doxygen/artifact_catalog.dox [hunk 4]
```diff
@@ -35,6 +35,23 @@ TSK_CARD_NUMBER (Credit card number)
...
+    or 
+  TSK_DATETIME
+
+### OPTIONAL ATTRIBUTES
+- TSK_ACTIVITY_TYPE (Activity Type i.e.: On or Off)
```
Comment (rcordovano) [inline bindings/java/doxygen/artifact_catalog.dox:48]: Type => type

File: bindings/java/doxygen/artifact_catalog.dox [hunk 5]
```diff
@@ -35,6 +35,23 @@ TSK_CARD_NUMBER (Credit card number)
...
+- TSK_DATETIME_END (When activity ended)
+- TSK_PROG_NAME (Name of the program doing the activity)
+- TSK_VALUE (Connection Type)
```
Comment (rcordovano) [inline bindings/java/doxygen/artifact_catalog.dox:51]: Type => type

File: bindings/java/doxygen/artifact_catalog.dox [hunk 6]
```diff
@@ -43,6 +60,20 @@ Provides a backwards link to an artifact that references the parent file of this
 - TSK_ASSOCIATED_ARTIFACT (Artifact ID of associated artifact)
 
 
+---
+## TSK_BACKUP
+Details about System/Aplication/File backups.
```
Comment (rcordovano) [inline bindings/java/doxygen/artifact_catalog.dox:65]: Aplication => Application

Maybe System/Application/File => System/application/file?

File: bindings/java/doxygen/artifact_catalog.dox [hunk 7]
```diff
@@ -556,6 +625,18 @@ Details about a remote drive found in the data source.
 
 
 
+---
+## TSK_SCREEN_SHOTS
+Screen shots from a device or Application.
```
Comment (rcordovano) [inline bindings/java/doxygen/artifact_catalog.dox:630]: Application => application

File: bindings/java/src/org/sleuthkit/datamodel/Bundle.properties [hunk 8]
```diff
@@ -55,6 +55,12 @@ BlackboardArtifact.tskUserContentSuspected.text=User Content Suspected
 BlackboardArtifact.tskMetadata.text=Metadata
 BlackboardArtifact.tskTrack.text=GPS Track
 BlackboardArtifact.tskWebAccountType.text=Web Account Type
+BlackboardArtifact.tskScreenShots.text=Screen Shots
+BlackboardArtifact.tskDhcpInfo.text=DHCP Information
+BlackboardArtifact.tskNotifications.text=Notification
```
Comment (rcordovano) [inline bindings/java/src/org/sleuthkit/datamodel/Bundle.properties:60]: Application => Applications ?

File: bindings/java/src/org/sleuthkit/datamodel/Bundle.properties [hunk 9]
```diff
@@ -55,6 +55,12 @@ BlackboardArtifact.tskUserContentSuspected.text=User Content Suspected
...
+BlackboardArtifact.tskBackup.text=Backup
```
Comment (rcordovano) [inline bindings/java/src/org/sleuthkit/datamodel/Bundle.properties:61]: Backup => Backups ?

File: bindings/java/src/org/sleuthkit/datamodel/Bundle.properties [hunk 10]
```diff
@@ -55,6 +55,12 @@ BlackboardArtifact.tskUserContentSuspected.text=User Content Suspected
 BlackboardArtifact.tskMetadata.text=Metadata
 BlackboardArtifact.tskTrack.text=GPS Track
 BlackboardArtifact.tskWebAccountType.text=Web Account Type
+BlackboardArtifact.tskScreenShots.text=Screen Shots
```
Comment (rcordovano) [inline bindings/java/src/org/sleuthkit/datamodel/Bundle.properties:58]: Screen Shots => Screenshots ?

**Review by rcordovano — CHANGES_REQUESTED** [review]

File: bindings/java/doxygen/artifact_catalog.dox (hunk 2 above)
Comment (markmckinnon) [reply bindings/java/doxygen/artifact_catalog.dox]: Fixed

File: bindings/java/doxygen/artifact_catalog.dox (hunk 3 above)
Comment (markmckinnon) [reply bindings/java/doxygen/artifact_catalog.dox]: Fixed

File: bindings/java/doxygen/artifact_catalog.dox (hunk 4 above)
Comment (markmckinnon) [reply bindings/java/doxygen/artifact_catalog.dox]: Fixed

File: bindings/java/doxygen/artifact_catalog.dox (hunk 5 above)
Comment (markmckinnon) [reply bindings/java/doxygen/artifact_catalog.dox]: Fixed

File: bindings/java/doxygen/artifact_catalog.dox (hunk 6 above)
Comment (markmckinnon) [reply bindings/java/doxygen/artifact_catalog.dox]: Fixed

File: bindings/java/doxygen/artifact_catalog.dox (hunk 7 above)
Comment (markmckinnon) [reply bindings/java/doxygen/artifact_catalog.dox]: Fixed

File: bindings/java/src/org/sleuthkit/datamodel/Bundle.properties (hunk 9 above)
Comment (markmckinnon) [reply bindings/java/src/org/sleuthkit/datamodel/Bundle.properties]: Fixed

File: bindings/java/src/org/sleuthkit/datamodel/Bundle.properties (hunk 10 above)
Comment (markmckinnon) [reply bindings/java/src/org/sleuthkit/datamodel/Bundle.properties]: Fixed

File: bindings/java/src/org/sleuthkit/datamodel/Bundle.properties (hunk 8 above)
Comment (markmckinnon) [reply bindings/java/src/org/sleuthkit/datamodel/Bundle.properties:60]: Not sure what you mean here.

Comment (rcordovano) [reply bindings/java/src/org/sleuthkit/datamodel/Bundle.properties:60]: Sorry, that was a copy and paste error. I am wondering if "Notification" should be "Notifications."

This is admittedly kind of awkward in a way that affects all of the artifact types. "Notification" makes sense as a type display name, but the primary use of these display names is in the main tree view under "Extracted Content," where a grouping tree node named "Notifications" makes more sense to me than one named "Notification." So I guess, yes, Notification => Notifications.

Comment (markmckinnon) [reply bindings/java/src/org/sleuthkit/datamodel/Bundle.properties]: Fixed

**Review by kellykelly3 — APPROVED** [review]

**Review by kellykelly3 — APPROVED** [review]

**Pull request merged**